apps/namada-interface/src/App/Token/EthereumBridge/EthereumBridge.tsx

```diff
@@ -0,0 +1,249 @@
+import { chains } from "@namada/chains";
+import { shortenAddress } from "@namada/utils";
+
+const SUPPORTED_TOKENS = ["TESTERC20", "NUTTESTERC20"];
+type FormFields = "recipient" | "amount" | "feeAmount";
+
+const toTokenData = (
+  accounts: Account[],
+  filterFn: (val: [tokenType: string, balance: BigNumber]) => boolean = () =>
+    true
+): Option<string>[] => {
+  return accounts.flatMap(({ balance, details }) => {
+    const { address, alias } = details;
+
+    return Object.entries(balance)
+      .filter(filterFn)
+      .map(([tokenType, amount]) => ({
+        value: `${address}|${tokenType}|${amount}`,
+        label: `${alias !== "Namada" ? alias + " - " : ""}${
+          Tokens[tokenType as TokenType].coin
+        } (${amount} ${tokenType})`,
+      }));
+  });
+};
+
+export const EthereumBridge = (): JSX.Element => {
+  const { derived } = useAppSelector<AccountsState>((state) => state.accounts);
+  const { chainId } = useAppSelector<SettingsState>((state) => state.settings);
+  const [recipient, setRecipient] = useState("");
+  const [amount, setAmount] = useState<BigNumber>(new BigNumber(0));
+  const [feeAmount, setFeeAmount] = useState<BigNumber>(new BigNumber(0));
+
+  const accounts = Object.values(derived[chainId]).filter(
+    ({ details }) => !details.isShielded
+  );
+
+  const transferableTokenData = toTokenData(accounts, ([tokenType]) =>
+    SUPPORTED_TOKENS.includes(tokenType)
+  );
+  const [token, setToken] = useState<string>(transferableTokenData[0]?.value);
+  const tokenBalance = BigNumber(token?.split("|").pop() || 0);
+  const tokenData = toTokenData(
+    accounts,
+    ([tokenType]) => !Boolean(Tokens[tokenType as TokenType]?.isNut)
+  );
+
+  const [feeToken, setFeeToken] = useState<string>(tokenData[0]?.value);
+  const feeTokenBalance = BigNumber(feeToken?.split("|").pop() || 0);
+
+  const [dirtyFields, setDirtyFields] = useState<Set<FormFields>>(new Set());
+
+  const handleSubmit = async (): Promise<void> => {
+    const [accountId, tokenSymbol] = token.split("|");
+    const [_, feeTokenSymbol] = feeToken.split("|");
+    const account = accounts.find(
+      (account) => account?.details?.address === accountId
+    ) as Account;
+
+    const integration = getIntegration(chainId);
+    await integration.submitBridgeTransfer(
+      {
+        bridgeProps: {
+          nut: Boolean(Tokens[tokenSymbol as TokenType]?.isNut),
+          tx: {
+            token: Tokens.NAM.address || "",
+            feeAmount: new BigNumber(0),
+            gasLimit: new BigNumber(20_000),
+            publicKey: account.details.publicKey,
+            chainId,
+          },
+          asset: Tokens[tokenSymbol as TokenType]?.nativeAddress || "",
+          recipient,
+          sender: account.details.address,
+          amount,
+          feeAmount,
+          feeToken: Tokens[feeTokenSymbol as TokenType]?.address || "",
+        },
+      },
+      //TODO: support ledger?
+      AccountType.Mnemonic
```

Comment:
I guess we _could_ set this to `account.details.type`, with the assumption that one day the Ledger will be able to sign that approval, but this is fine for now!